test/handler.test.js

```
@@ -361,6 +361,130 @@ describe('strong-error-handler', function() {
+      // Mock an error reported by fs.readFile
+      var error = new ErrorWithProps({
+        name: 'Error',
+        message: 'ENOENT: no such file or directory, open "/etc/passwd"',
+        errno: -2,
+        code: 'ENOENT',
+        syscall: 'open',
+        path: '/etc/password',
+      });
+      givenErrorHandlerForError(error);
+
+      requestHTML()
+        .end(function(err, res) {
+          expect(res.statusCode).to.eql(500);
+          var body = res.error.text;
+          expect(body).to.not.match(/\/etc\/password/);
+          expect(body).to.not.match(/-2/);
+          expect(body).to.not.match(/ENOENT/);
+          // only have the following
+          expect(body).to.match(/<title><\/title>/);
+          expect(body).to.match(/500(.*?)Internal Server Error/);
+          done();
+        });
+    });
+
+    function requestHTML(url) {
+      return request.get(url || '/')
+        .set('Accept', 'text/html')
+        .expect('Content-Type', /^text\/html/);
+    }
+  });
+
+  context('Content Negotiation', function() {
+    it('should honor default to json without options', function(done) {
+      givenErrorHandlerForError(new Error('Some error'), {});
+      request.get('/')
+        .set('Accept', '*/*')
+        .expect('Content-Type', /^application\/json/, done);
+    });
+
+    it('should honor accepted content-type', function(done) {
+      givenErrorHandlerForError(new Error('Some error'), {
+        defaultType: 'application/json',
+      });
+      request.get('/')
+        .set('Accept', 'text/html')
+        .expect('Content-Type', /^text\/html/, done);
+    });
+
+    it('considers order of Accepted content-types', function(done) {
+      givenErrorHandlerForError(new Error('Some error'), {});
+      request.get('/')
+        // `*/*` will be used because its provided first
+        .set('Accept', '*/*, text/html')
+        .expect('Content-Type', /^application\/json/, function() {
+          request.get('/')
```

Please split this into two tests, each test making a single request.

Your current implementation ignores errors reported by the first `request.get...expect`.